netlogo-gui/src/main/agent/Torus3D.java

```diff
@@ -8,51 +8,59 @@
 public final strictfp class Torus3D
     extends Torus
     implements Topology3D {
+
+  // this class *ought* to be able to see `world` in Topology, but
+  // it can't, probably due to the scala class `Torus` between this
+  // class and Topology. in the meantime, adding this field is good enough
+  // for now.
+  final World _world;
```

Comment:
ah. this seems confused/confusing enough that filing a bug certainly might be appropriate, but probably isn't a good use of your time.